lib/_http_client.js

```diff
@@ -636,7 +636,11 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 function onSocketNT(req, socket) {
   if (req.aborted) {
     // If we were aborted while waiting for a socket, skip the whole thing.
-    socket.emit('free');
+    if (req.socketPath || !req.agent) {
+      socket.destroy();
+    } else {
+      socket.emit('free');
+    }
```

Comment:
This is when `socket.emit('free');` was added: https://github.com/nodejs/node/commit/d0c7d93536d8bc7977ed2d3bd0b0246a558e5137#diff-e3bc37430eb078ccbafe3aa3b570c91a, previously the socket was destroyed in all cases.